src/gameTest/java/mekanism/common/tests/energy/EnergyConversionTest.java

```diff
@@ -0,0 +1,237 @@
+import mekanism.common.tests.helpers.EnergyTestHelper;
+import net.minecraft.gametest.framework.GameTest;
+import net.neoforged.neoforge.energy.EnergyStorage;
+import net.neoforged.neoforge.energy.IEnergyStorage;
+import net.neoforged.testframework.annotation.ForEachTest;
+import net.neoforged.testframework.annotation.TestHolder;
+import net.neoforged.testframework.gametest.EmptyTemplate;
+
+@ForEachTest(groups = "energy.conversion")
+public class EnergyConversionTest {
+
+    @GameTest
+    @EmptyTemplate
+    @TestHolder(description = "Tests that energy conversion works properly with various decimal values.")
+    public static void testForgeEnergy(final EnergyTestHelper helper) {
+        double defaultConversion = 2.5;
+        double inverseConversion = 1 / defaultConversion;
+        int capacity = 1_000;
+        int convertedCapacity = (int) (capacity / defaultConversion);
+        helper.startSequence()
+              //Ensure the config starts out at the default value
+              .thenExecute(() -> MekanismConfig.general.forgeConversionRate.set(defaultConversion))
+
+              // WRAPPING STRICT ENERGY TO FORGE ENERGY
+
+              //Test against a small empty container
+              .thenMap(() -> helper.createForgeWrappedStrictEnergyHandler(0, capacity))
+              .thenExecute(handler -> {
+                  int extracted = handler.extractEnergy(capacity, false);
+                  helper.assertValueEqual(extracted, 0, "extracted energy");
+                  int accepted = handler.receiveEnergy(capacity, false);
+                  helper.assertValueEqual(accepted, convertedCapacity, "accepted energy");
+                  helper.assertValueEqual(handler.getEnergyStored(), convertedCapacity, "stored energy");
+              })
+              //Test against a small full container
+              .thenMap(() -> helper.createForgeWrappedStrictEnergyHandler(capacity, capacity))
+              .thenExecute(handler -> {
+                  int accepted = handler.receiveEnergy(capacity, false);
+                  helper.assertValueEqual(accepted, 0, "accepted energy");
+                  int extracted = handler.extractEnergy(capacity, false);
+                  helper.assertValueEqual(extracted, convertedCapacity, "extracted energy");
+                  helper.assertValueEqual(handler.getEnergyStored(), 0, "stored energy");
+              })
+              //Test inserting against a small nearly full container
+              .thenMap(() -> helper.createForgeWrappedStrictEnergyHandler(capacity - 2, capacity))
+              .thenExecute(handler -> {//There shouldn't be any room for it
+                  int accepted = handler.receiveEnergy(capacity, false);
+                  helper.assertValueEqual(accepted, 0, "accepted energy");
+                  //Note: Needed energy should be 1 even though we can't accept it
+                  helper.assertValueEqual(handler.getEnergyStored(), convertedCapacity - 1, "stored energy");
```

Comment:
I would have done this the other way. Calculation should go in the first argument, expected result should be `1`